hoomd/operations.py

```diff
@@ -75,38 +152,57 @@ def _store_reader(self, reader):
         # TODO
         pass
 
-    def __contains__(self, obj):
-        return any(op is obj for op in self)
+    def __contains__(self, operation):
+        """Whether an operation is contained in this container.
+
+        Args:
+            operation (``any``): Returns whether this exact operation is
+                contained in the collection.
+        """
+        return any(op is operation for op in self)
 
     def __iter__(self):
-        if self._integrator is not None:
-            yield from chain((self._integrator,), self._analyzers,
-                             self._updaters, self._tuners)
-        else:
-            yield from chain((self._analyzers, self._updaters, self._tuners))
+        """Iterates through all contained operations."""
+        integrator = (self._integrator,) if self._integrator else []
+        yield from chain(
+            integrator, self._analyzers, self._updaters, self._tuners)
+
+    def __len__(self):
+        """Return the number of operations contained in this collection."""
+        return len(list(self))
 
     @property
     def scheduled(self):
+        """bool: Whether `Operations.schedule` has been called and is in
+        effect.
+        """
         return self._scheduled
 
     @property
     def integrator(self):
+        """``Integrator``: An MD or HPMC integrator object.
```

Comment:
Is this formatted correctly? The other Sphinx class references use one backtick, I believe. Unless this is supposed to be a reference to a private base class like `_BaseIntegrator` (see comment below about documenting private base types, it seems that @joaander and I agree this is a good idea).

Reply:
Yes, we discussed this on Monday. We will document these core base classes project-wide in a separate PR.